pkg/client/apify_client.go

```diff
@@ -0,0 +1,259 @@
+	if err != nil {
+		logrus.Errorf("error making GET request: %v", err)
+		return nil, fmt.Errorf("error making GET request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logrus.Errorf("error reading response body: %v", err)
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	// Check response status
+	if resp.StatusCode != http.StatusOK {
+		logrus.Errorf("unexpected status code %d: %s", resp.StatusCode, string(body))
+		return nil, fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(body))
+	}
+
+	// Parse response
+	var runResp ActorRunResponse
+	if err := json.Unmarshal(body, &runResp); err != nil {
+		logrus.Errorf("error parsing response: %v", err)
+		return nil, fmt.Errorf("error parsing response: %w", err)
+	}
+
+	return &runResp, nil
+}
+
+// GetDatasetItems gets items from a dataset with pagination
+func (c *ApifyClient) GetDatasetItems(datasetId string, offset, limit int) (*DatasetResponse, error) {
+	url := fmt.Sprintf("%s/datasets/%s/items?token=%s&offset=%d&limit=%d",
+		c.baseUrl, datasetId, c.apiToken, offset, limit)
+	logrus.Debugf("Getting dataset items: %s (offset: %d, limit: %d)", datasetId, offset, limit)
+
+	// Create request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logrus.Errorf("error creating GET request: %v", err)
+		return nil, fmt.Errorf("error creating GET request: %w", err)
+	}
+
+	// Make the request
+	resp, err := c.options.HttpClient.Do(req)
+	if err != nil {
+		logrus.Errorf("error making GET request: %v", err)
+		return nil, fmt.Errorf("error making GET request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logrus.Errorf("error reading response body: %v", err)
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	// Check response status
+	if resp.StatusCode != http.StatusOK {
+		logrus.Errorf("unexpected status code %d: %s", resp.StatusCode, string(body))
+		return nil, fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(body))
+	}
+
+	// Parse response - Apify returns a direct array of items, not wrapped in a data object
+	var items []json.RawMessage
+	if err := json.Unmarshal(body, &items); err != nil {
+		logrus.Errorf("error parsing response: %v", err)
+		return nil, fmt.Errorf("error parsing response: %w", err)
+	}
+
+	// Create a DatasetResponse object with the items and estimated pagination info
+	datasetResp := &DatasetResponse{
+		Data: struct {
+			Items  []json.RawMessage `json:"items"`
+			Count  int               `json:"count"`
+			Offset int               `json:"offset"`
+			Limit  int               `json:"limit"`
+			Total  int               `json:"total"`
+		}{
+			Items:  items,
+			Count:  len(items),
+			Offset: offset,
+			Limit:  limit,
+			Total:  offset + len(items), // Estimate total, could be more if limit is reached
```

Comment:
I don't get this `Total` calculation. Say that there are actually 40 total items and Limit=10. On the first call, `offset` is 0 and `limit` is 10, so it would say the total is 10. On the next one, you get a total of 20, and so on and so forth. Or am I reading this wrong?

Reply:
ah you are right, total is a relic from a different approach that was never cleaned up 😓